src/components/pages/txs/fetch-transactions.ts

```diff
@@ -0,0 +1,203 @@
+import { range } from "lodash";
+import { Hash } from "viem";
+import { prisma } from "@/lib/prisma";
+import {
+  TransactionWithReceipt,
+  fromPrismaTransactionWithReceipt,
+  fromViemTransactionWithReceipt,
+} from "@/lib/types";
+import { loadFunctions } from "@/lib/signatures";
+import { l2PublicClient } from "@/lib/chains";
+
+const fetchLatestTransactionsFromDatabase = async (
```

Comment:
Your database implementation is wrong as it tries to replicate exactly the JSON-RPC implementation.

The JSON-RPC implementation is a little bit circonvoluted as there's no easy way by default to fetch transactions with some kind of pagination in mind, so we vastly overfetch blocks from before and after the `start`ing block, build an array to help with pagination and only fetch the transactions and receipts of the current page. 

With the indexer and the database however, we can directly target the transactions we're looking for using their `blockNumber` and `transactionIndex`.
The correct way of fetching the transactions should be querying the transactions model directly and making sure the `blockNumber` is within range (use the same range as JSON-RPC version), order by `blockNumber` and `transactionIndex` and you will get the same pagination-friendly array. You can limit the selected fields to only include whats necessary in this step, eg. tx hash, tx index, block number and tx input to load the signatures immediately after.
Then use the same logic as the JSON-RPC version to get the current page items and perform a second prisma query with all the fields selected and the receipt included to build the final result.

Reply:
I got it sir! Np, I'll update my implementation following your guidance, thank you!

Reply:
I added some changes, pagination seems pretty smooth, and now I'm fetching the transactions instead of the blocks